velox/dwio/parquet/writer/Writer.cpp

```diff
@@ -161,13 +176,15 @@ Writer::Writer(
 
 Writer::Writer(
     std::unique_ptr<dwio::common::FileSink> sink,
-    const WriterOptions& options)
+    const WriterOptions& options,
+    const RowTypePtr& schema)
     : Writer{
           std::move(sink),
           options,
           options.memoryPool->addAggregateChild(fmt::format(
               "writer_node_{}",
-              folly::to<std::string>(folly::Random::rand64())))} {}
+              folly::to<std::string>(folly::Random::rand64()))),
+          std::move(schema)} {}
```

Comment:
Can you check schema here as well? Perhaps, extract a helper function for checking schema to avoid copy-paste.